engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h

```diff
@@ -259,39 +260,38 @@ struct tree_processor {
 		print_tree(&this->root);
 		printf("==============================================================\n");
 
-		ral::cache::graph graph;
 		if (this->root.kernel_unit != nullptr) {
-			graph.add_node(this->root.kernel_unit.get()); // register first node
-			visit(graph, &this->root, this->root.children);
+			query_graph->add_node(this->root.kernel_unit.get()); // register first node
+			visit(*query_graph, &this->root, this->root.children);
 		}
-		return graph;
+		return query_graph;
 	}
 
-	void visit(ral::cache::graph& graph, node * parent, std::vector<std::shared_ptr<node>>& children) {
+	void visit(ral::cache::graph& query_graph, node * parent, std::vector<std::shared_ptr<node>>& children) {
```

Comment:
what is the intention of this funciton, the signature is very questionable and liable to cause all kinds of spectacular failures.

Reply:
i am not sure. I think its part of a visitor pattern when creating the graph. Here I just changed the name of the variable because i was getting compilation errors since the name of the variable was exactly the same as the name of the class. Sometimes that is ok, sometimes it confuses the hell out of the compiler and the developer debugging why it wont build

Reply:
shouldn't matter unless omsone put a "using" clause. this function should be discussed in a call at some point to address my concerns i guess they arent relevant to this PR